moveit_core/kinematic_constraints/src/pykinematic_constraint.cpp

```diff
@@ -0,0 +1,45 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include <pybind11/eigen.h>
+#include <pybind11/stl.h>
+#include <moveit/python/pybind_rosmsg_typecasters.h>
+
+#include <moveit/kinematic_constraints/kinematic_constraint.h>
+#include <moveit/kinematic_constraints/utils.h>
+
+namespace py = pybind11;
+
+using namespace kinematic_constraints;
+
+auto CONSTRUCT_GOAL_CONSTRAINTS1 = py::overload_cast<const std::string&,                 //
+                                                     const geometry_msgs::PoseStamped&,  //
+                                                     double,                             //
+                                                     double>(&constructGoalConstraints);
+
+auto ADD1 = py::overload_cast<const moveit_msgs::Constraints&,  //
+                              const moveit::core::Transforms&>(&KinematicConstraintSet::add);
+
+auto DECIDE1 = py::overload_cast<const moveit::core::RobotState&, bool>(&KinematicConstraintSet::decide, py::const_);
+
+void def_kinematic_constraints_bindings(py::module& m)
+{
+  m.doc() = "Class for joint, position, visibility, and other constraints";
+  m.def("constructGoalConstraints",
+        CONSTRUCT_GOAL_CONSTRAINTS1,      //
```

Comment:
Same here as above: Please embed the `py::overload_cast` directly. Otherwise, it doesn't become clear immediately, where the extra definitions belong to.

Reply:
Sure I can do that. My reason for pulling them out was formatting. It becomes pretty unreadable with them defined inline